plugins/gaming.py

```diff
@@ -116,19 +116,74 @@ def dice(text, notice):
         return "{} ({})".format(total, ", ".join(rolls))
 
 
+choice_split_re = re.compile(r'\s*(?:, ?or|,|or)\s*', re.I)  # regex for splitting choices for choose command
+
+
 @hook.command
 def choose(text, event):
     """<choice1>, [choice2], [choice3], etc. - randomly picks one of the given choices
     :type text: str
     """
-    choices = re.findall(r'([^,]+)', text.strip())
+    choices = choice_split_re.split(text.strip())
+
     if len(choices) == 1:
-        choices = choices[0].split(' or ')
-        if len(choices) == 1:
+        event.notice_doc()
+        return
+
+    return random.choice([choice.strip() for choice in choices])
+
+
+# array of digits/characters indexed by int value
+val_char_map = [chr(x) for x in
+                [c for r in (range(ord('0'), ord('9') + 1), range(ord('a'), ord('z') + 1)) for c in r]]
+
+
+def order_bounds(value_a, value_b):
+    """Return the input bounds correctly ordered for randint function"""
+    if value_a < value_b:
+        return value_a, value_b
+
+    return value_b, value_a
+
+
+@hook.command("chooser", "chooserange")
+def chooserange(text, event, notice):
+    """<a> to <b>|<a> <b> - picks a random integer or character from the given range"""
+    choices = re.split(r'\s*(?:to| +)\s*', text)
+
+    if len(choices) == 2:
+        try:
+            try:
+                # attempt to parse as integer range
+                lower_bound, upper_bound = order_bounds(int(choices[0]), int(choices[1]))
+
+                return str(random.randint(lower_bound, upper_bound))
+
+            except ValueError:
```

Comment:
A `try-except` should be limited to cover only the relevant code, in this case the `int()` conversions. Please split those conversions out on to their own line(s) and have the `try-except` cover those.